sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/TrainUnlabeledCustomModelAsync.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.formrecognizer;
+
+import com.azure.ai.formrecognizer.models.CustomFormModel;
+import com.azure.ai.formrecognizer.models.OperationResult;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.PollerFlux;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This sample demonstrates how to train a model with unlabeled data. See RecognizeCustomFormsAsync
+ * to recognize forms with your created custom model.
+ */
+public class TrainUnlabeledCustomModelAsync {
+
+    /**
+     * Main method to invoke this demo.
+     *
+     * @param args Unused arguments to the program.
+     */
+    public static void main(String[] args) {
+        // Instantiate a client that will be used to call the service.
+
+        FormTrainingAsyncClient client = new FormRecognizerClientBuilder()
+            .apiKey(new AzureKeyCredential("{api_key}"))
+            .endpoint("https://{endpoint}.cognitiveservices.azure.com/")
+            .buildAsyncClient().getFormTrainingAsyncClient();
+
+        // Train custom model
+        String trainingSetSource = "{training_set_SAS}";
+        PollerFlux<OperationResult, CustomFormModel> trainingPoller = client.beginTraining(trainingSetSource, false);
+
+        Mono<CustomFormModel> customFormModelResult = trainingPoller
+            .last()
```

Comment:
is possible to just use .subscribe() instead of .last()?

Reply:
.last() definitely gives better control over the code flow to not continue until a response is received.